dockerfiles/compile.go

```diff
@@ -46,6 +49,7 @@ func applyTemplate(template string, arguments templateArguments) string {
 		"{package_build}":     arguments.package_build,
 		"{tar_distro_name}":   arguments.tar_distro_name,
 		"{package_extension}": arguments.package_extension,
+		"{go_version}":        arguments.go_version,
```

Comment:
Do we want to instead add an `ARG GO_VERSION=1.21.1` around line 27 in the `Dockerfile` to make future upgrade diffs smaller?

Reply:
That's probably fine I can do that. Want to let this CI run finish first then will push that